pydantic/networks.py

```diff
@@ -176,6 +181,18 @@ def build(
         fragment: Optional[str] = None,
         **_kwargs: str,
     ) -> str:
+        parts = Parts(
```

Comment:
not sure why this is required and/or part of this change?

Reply:
This is necessary because `get_default_parts` (there the default port value is checked) and `build` are both class methods, and they cannot share any information using `self` object. Also `get_default_args` does not support kwargs, only the `Parts` class object, so I need to create it manually.